### PR TITLE
feat(aws,gce) render entity tags

### PR DESCRIPTION
**Pull request by danielpeach**
@icfantv @anotherchrisberry please review.

- Renders AWS server group entity tags outside of Netflix mode.
- Adds feature flags to all the entity tag components.

### CODE REVIEW

**Comment by danielpeach** [comment]
I added in the GCE bits because it required minimal work. @duftler please review.

File: app/scripts/modules/core/utils/renderIfFeature.directive.ts [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+class RenderIfFeatureDirectiveController {
+  public $attrs: ng.IAttributes;
+  public $element: JQuery;
+
+  constructor(private settings: any) {}
+
+  public initialize(): void {
+    let feature: string = this.$attrs['renderIfFeature'];
+    if (feature && !this.settings.feature[feature]) {
+      this.$element.addClass('ng-hide');
```
Comment (icfantv) [inline app/scripts/modules/core/utils/renderIfFeature.directive.ts]: just a thought, this strikes me as a strange way to hide the element.  can we not just use the `hidden` class, or one of the others, from bootstrap?

Comment (danielpeach) [inline app/scripts/modules/core/utils/renderIfFeature.directive.ts]: Yep, that's reasonable. Thanks.

File: app/scripts/modules/core/utils/renderIfFeature.directive.ts [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+class RenderIfFeatureDirectiveController {
+  public $attrs: ng.IAttributes;
+  public $element: JQuery;
+
+  constructor(private settings: any) {}
+
+  public initialize(): void {
+    let feature: string = this.$attrs['renderIfFeature'];
+    if (feature && !this.settings.feature[feature]) {
+      this.$element.addClass('hidden');
```
Comment (anotherchrisberry) [inline app/scripts/modules/core/utils/renderIfFeature.directive.ts]: What happens if you try to remove the element altogether? My only concern is that the element will still be rendered and run through the Angular lifecycle, and that may result in surprising behavior if some things are only enabled by the feature.

File: app/scripts/modules/core/utils/renderIfFeature.directive.ts [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+import {DirectiveFactory} from './tsDecorators/directiveFactoryDecorator';
+
+class RenderIfFeatureDirectiveController {
+  public $attrs: ng.IAttributes;
+  public $element: JQuery;
+
+  constructor(private settings: any) {}
+
+  public initialize(): void {
+    let feature: string = this.$attrs['renderIfFeature'];
```
Comment (anotherchrisberry) [inline app/scripts/modules/core/utils/renderIfFeature.directive.ts]: We just updated to TS 2.2, which lets you specify this as `this.$attrs.renderIfFeature` (I think) - can you try that and see if the compiler complains?

File: app/scripts/modules/core/utils/renderIfFeature.directive.ts [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+}
+
+@DirectiveFactory('settings')
+class RenderIfFeatureDirective implements ng.IDirective {
+  public restrict = 'A';
+  public controller: any = RenderIfFeatureDirectiveController;
+  public controllerAs = '$ctrl';
+
+  public link($scope: ng.IScope, $element: JQuery, $attrs: ng.IAttributes) {
+    let ctrl: RenderIfFeatureDirectiveController = $scope['$ctrl'];
```
Comment (anotherchrisberry) [inline app/scripts/modules/core/utils/renderIfFeature.directive.ts]: same as above (`$scope.$ctrl` might work now)

File: app/scripts/modules/core/utils/renderIfFeature.directive.ts (hunk 2 above)
Comment (danielpeach) [inline app/scripts/modules/core/utils/renderIfFeature.directive.ts]: That is a very good point.

Comment (anotherchrisberry) [inline app/scripts/modules/core/utils/renderIfFeature.directive.ts]: You might get into some weird priority stuff with the directive firing and initializing after the rest of the component, so it'll be a really fun thing for you to work through.

**Pull request merged**